pkg/azure/plugin.go

```diff
@@ -847,6 +847,211 @@ func (s *azurePluginServer) AttachResource(ctx context.Context, attachResourceRe
+	}
+
+	checkResponse.Resource = &paragliderpb.ResourceInfo{}
+	checkResponse.Resource.Name = *resource.Name
+	checkResponse.Resource.Uri = *resource.ID
+	checkResponse.Resource.Ip = resourceIp
+	checkResponse.MissingResources = missingRescources
+
+	return checkResponse, nil
+}
+
+func (s *azurePluginServer) CheckPermitLists(ctx context.Context, handler *AzureSDKHandler, resourceID string, networkInfo *resourceNetworkInfo, namespace string, shouldFix bool) ([]*paragliderpb.MissingResource, error) {
+	vnetName := getVnetFromSubnetId(networkInfo.SubnetID)
+	vnet, err := handler.GetVirtualNetwork(ctx, vnetName)
+	if err != nil {
+		utils.Log.Printf("An error occured while getting vnet:%+v", err)
+		return nil, err
+	}
+
+	// Get subnets address spaces
+	localVnetAddressSpaces := []string{}
+	for _, addressSpace := range vnet.Properties.AddressSpace.AddressPrefixes {
+		localVnetAddressSpaces = append(localVnetAddressSpaces, *addressSpace)
+	}
+	if len(localVnetAddressSpaces) == 0 {
+		return nil, fmt.Errorf("unable to get subnet address prefix")
+	}
+
+	missingResources := []*paragliderpb.MissingResource{}
+	visitedIps := map[string]bool{}
+	deletedIps := map[string]bool{}
+
+	requireNAT := false
+	// Get used address spaces of all clouds
+	orchestratorConn, err := grpc.NewClient(s.orchestratorServerAddr, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		return nil, fmt.Errorf("unable to establish connection with orchestrator: %w", err)
+	}
+	defer orchestratorConn.Close()
+	orchestratorClient := paragliderpb.NewControllerClient(orchestratorConn)
+	getUsedAddressSpacesResp, err := orchestratorClient.GetUsedAddressSpaces(context.Background(), &emptypb.Empty{})
+	if err != nil {
+		return nil, fmt.Errorf("unable to get used address spaces: %w", err)
+	}
+
+	permitListReq := &paragliderpb.GetPermitListRequest{Namespace: namespace, Resource: resourceID}
+	permitListResp, err := s.GetPermitList(ctx, permitListReq)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, rule := range permitListResp.Rules {
+		peeringCloudInfos, err := utils.GetPermitListRulePeeringCloudInfo(rule, getUsedAddressSpacesResp.AddressSpaceMappings)
+		if err != nil {
+			return nil, fmt.Errorf("unable to get peering cloud infos: %w", err)
+		}
+
+		for i, peeringCloudInfo := range peeringCloudInfos {
+			peerIp := rule.Targets[i]
+
+			// If we're just checking, we don't need to check the same IP twice
+			// if it exists for both inbound and outbound rules
+			// If fixing, both inbound and outbound rules need to be deleted
+			if !shouldFix && visitedIps[peerIp] {
+				continue // Skip if the IP has already been visited and checked
+			}
+
+			if peeringCloudInfo == nil {
+				requireNAT = true
+			} else if peeringCloudInfo.Cloud == utils.AZURE {
+				// Ips don't can't be repeated, so if a deleted Ip is encountered again
+				// in a diff. rule, it's for conn. in the opposite direction and should be deleted
```

Comment:
Not sure I understand this. Why would that imply its in the opposite direction?

Reply:
Inbound target : [10.1.1.1]
Outbound target: [10.1.1.1]

Since we don't support overlapping addresses, any 10.1.1.1 references the same resource (regardless of cloud). If the target's inbound rule has been deleted, the target's outbound rule should be deleted as well